### ISSUE TITLE
Making an api mock without starting a real server

### ISSUE THREAD

**Issue by poloeater** [Nykredit/gript#68]
We would like to be able to mock the services called by the client to get the same effect as the mocked server functionality already in Gript.
However, to be able to use the mocked api on the auto created feature client deploys, we need the mock to be a part of the app itself instead of a running server.

I have looked into achieving this and the most apparent ways would be:
angular-multimocks
ngMockE2E

We need the switching of the mock to be pretty easy and either be two different gulp targets (like the mock server you proposed) or a runtime configuration, which you for instance could set through the query-url (like ?api-mock=true).

**Comment by jotka** [comment]
I'm going to include [Multimocks](https://github.com/wongatech/angular-multimocks).
Since there is [Stubby](https://github.com/mrak/stubby4node) already present in the workflow, I'm going to create additional configuration section in the `gulpfile.js`:

`mockEngines: ['stubby', multimocks']`

You will have an option to start both of them, just a single mock engine or none.

[Stubby](https://github.com/mrak/stubby4node) will be executed when doing `gulp` or `gulp server` as it is today. This still can be handy for someone.
[Multimocks](https://github.com/wongatech/angular-multimocks), when turned on, will generate it's files when doing `build` or `dist`.

FYI @hindsholm @radzikp

**Comment by jotka** [comment]
### implemented in 0.0.50

There are two mock engines implementations included in Gript:  [stubby4node](https://github.com/mrak/stubby4node) and [Angular-Multimocks](https://github.com/wongatech/angular-multimocks). [stubby4node](https://github.com/mrak/stubby4node) allows you to have a mock server running during the development, using the standard gulp or gulp server tasks. [Angular-Multimocks](https://github.com/wongatech/angular-multimocks), on the other hand - will generate some source code which will enable mocking services within your application (useful when you want to deploy your application with mocked services). You setup the mock configuration in the `mocks` section in the `gulpfile.js`:

```
mocks: {
    stubby: {
        location: 'localhost',
        stubs: 8050,
        tls: 8443,
        admin: 8051,
        relativeFilesPath: true,
        files: [
            'mocks/*.{json,yaml,js}'
        ]
    },
    multimocks: {
        src: 'multiMocks'
    }
}

```
You have an option to start both of them, just a single one or none. If the configuration for the specific mock engine is not present in the `gulpfile.js` - it will not be started.

[Angular-Multimocks](https://github.com/wongatech/angular-multimocks) lets you test how your app behaves with different responses from an API. Angular Multimocks allows you to define sets of mock API responses for different scenarios as JSON files.
The only mandatory configuration option exposed for Angular-multimocks is the `src`: the directory to load mock files from.
Gript will read your mock files and generate the needed source files (mock scenarios) into the `multimocks.js` file in the `target/tmp/js` folder. They will be minified and concatenated with your other JS files when doing `dist`.
Take note that if you want [Angular-Multimocks](https://github.com/wongatech/angular-multimocks) to do it's job, you need to include `angular-multimocks` and `Angular Mocks` (which [Angular-Multimocks](https://github.com/wongatech/angular-multimocks) depends on) in your `bower.rc`:

```
 "angular-mocks": "1.2.29",
 "angular-multimocks": "0.6.8",
```
The `sample_configs` contains `multiMocks` directory with example `mockResources.json` and some mock responses to get you started.
Refer to the [Angular-Multimocks](https://github.com/wongatech/angular-multimocks) documentation for more details.

**Comment by jotka** [comment]
@poloeater the "mocks / no mocks" switching needs to be done in your application. It's about including the module `scenario` in your application dependencies:

```
angular
  .module('demo', ['scenario'])
  // more code here...
```
